crates/xilem_web/src/interfaces.rs

```diff
@@ -91,11 +91,45 @@ where
         }
     }
 
-    // TODO should some methods extend some properties automatically,
-    // instead of overwriting the (possibly set) inner value
-    // or should there be (extra) "modifier" methods like `add_class` and/or `remove_class`
-    fn class(self, class: impl Into<Cow<'static, str>>) -> Attr<Self, T, A> {
-        self.attr("class", class.into())
+    /// Add a class to the wrapped element.
+    ///
+    /// If multiple classes are added, all will be applied to the element.
+    fn class(self, class: impl Into<Cow<'static, str>>) -> Class<Self, T, A> {
+        self.class_opt(Some(class))
+    }
+
+    /// Add an optional class to the wrapped element.
+    ///
+    /// If multiple classes are added, all will be applied to the element.
+    fn class_opt(self, class: Option<impl Into<Cow<'static, str>>>) -> Class<Self, T, A> {
+        Class {
+            element: self,
+            class_name: class.map(Into::into),
+            phantom: PhantomData,
+        }
+    }
+
+    /// Set a style attribute
+    fn style(
```

Comment:
Same here. I could imagine something like this `el.style([("position", "absolute"), ("background", "black")])` or similar as this method `el.style(("position", "absolute"))`.

But yeah I'm generally not sure about the slightly added complexity that a trait introduces to achieve that syntax sugar. But it would e.g. allow custom structs to implement that style trait.

Reply:
I think it's worth it because the alternative is more wrapper structs (`Class<Class<Style<Style<Style<Attr<...<Element>..>>>>>`)

Reply:
So you'd probably have `T`, `Option<T>`, `Vec<T>` perhaps for `T: impl Into<Cow<'static, str>>`

Reply:
I could imagine even more, but something along those lines yeah. Btw. you're talking about `class`? As style needs key-values? The array/container types could act directly on the trait like `Vec<T> where T: IntoClass`. But I'm not sure about stuff like `el.style((("position", "absolute"), CustomStyle::new()))` which is probably difficult to achieve via blanket-impls. 

Another idea regarding styling I have, is to use an enum for more structured styling, Something like:

```rust
enum StyleAttr {
    Background(BackgroundStyle),
    Position(PositionStyle),
} 
```
etc.

But this could be quite a lot of boilerplate to maintain (including the relevant syntax-sugar logic via traits), so not sure about that... Maybe we should iterate slowly towards more static/structured ways to achieve it (while always comparing whether it improves the current state)?